Sources/XCTestDynamicOverlay/XCTUnimplemented.swift

```diff
@@ -379,12 +394,58 @@ private func _unimplementedFatalError(_ message: String, file: StaticString, lin
   )
 }
 
+private enum Witness<Value> {}
 protocol AnyRangeReplaceableCollection {
   static func empty() -> Any
 }
-private enum Witness<Value> {}
 extension Witness: AnyRangeReplaceableCollection where Value: RangeReplaceableCollection {
   static func empty() -> Any {
     Value()
   }
 }
+protocol AnyExpressibleByArrayLiteral {
+  static func empty() -> Any
+}
+extension Witness: AnyExpressibleByArrayLiteral where Value: ExpressibleByArrayLiteral {
+  static func empty() -> Any {
+    [] as Value
+  }
+}
+protocol AnyExpressibleByDictionaryLiteral {
+  static func empty() -> Any
+}
+extension Witness: AnyExpressibleByDictionaryLiteral where Value: ExpressibleByDictionaryLiteral {
+  static func empty() -> Any {
+    [:] as Value
+  }
+}
+protocol AnyFixedWidthInteger {
+  static func zero() -> Any
+}
+extension Witness: AnyFixedWidthInteger where Value: FixedWidthInteger {
+  static func zero() -> Any {
+    Value.zero
+  }
+}
+protocol AnyBinaryFloatingPoint {
+  static func zero() -> Any
+}
+extension Witness: AnyBinaryFloatingPoint where Value: BinaryFloatingPoint {
+  static func zero() -> Any {
+    Value.zero
+  }
+}
+
+protocol DefaultConstructible {
+  init()
+}
+extension Bool: DefaultConstructible {}
+extension Set: DefaultConstructible {}
+extension Dictionary: DefaultConstructible {}
```

Comment:
These are a bit redundant as they are cover by `AnyExpressibleByArrayLiteral` and `AnyExpressibleByDictionaryLiteral`